app/script/auth/page/Login.js

```diff
@@ -116,11 +116,23 @@ class Login extends React.PureComponent {
 
   componentDidMount = () => {
     this.props.resetError();
+    const immediateLogin = URLUtil.hasURLParameter(QUERY_KEY.IMMEDIATE_LOGIN);
+    if (immediateLogin) {
+      return this.immediateLogin();
+    }
     this.readAndUpdateParamsFromUrl();
   };
 
   componentWillReceiveProps = nextProps => this.readAndUpdateParamsFromUrl(nextProps);
 
+  immediateLogin = () => {
+    return Promise.resolve()
+      .then(() => this.props.doInit({isImmediateLogin: true}))
+      .then(() => this.props.doInitializeClient(true, undefined))
+      .then(() => window.location.replace(URLUtil.pathWithParams(EXTERNAL_ROUTE.WEBAPP)))
```

Comment:
@Yserz I suppose we should also check that the user has set a username in this case instead of immediately redirecting to the webapp.